### PR TITLE
[Mac] Add new BasePropertyControlTextField to have ellipses and Allow…

### PR DESCRIPTION
**Pull request by CartBlanche**
…sExpansionToolTips by default.

Fixes: #531 

screen shot 2019-02-13 at 14 53 39

### CODE REVIEW

File: Xamarin.PropertyEditing.Mac/Controls/Custom/NumericTextField.cs [hunk 1]
```diff
@@ -93,9 +93,8 @@ public static bool CheckIfNumber (string finalString, ValidationType mode, bool
 
 		public static bool ValidateDecimal (string finalString, bool allowNegativeValues)
 		{
-			double value;
 			//Checks parsing to number
-			if (!double.TryParse (finalString, System.Globalization.NumberStyles.Any, System.Globalization.CultureInfo.CurrentUICulture, out value))
+			if (!double.TryParse (finalString, System.Globalization.NumberStyles.Any, System.Globalization.CultureInfo.CurrentUICulture, out var value))
```
Comment (ermau) [inline Xamarin.PropertyEditing.Mac/Controls/Custom/NumericTextField.cs]: Going forward, let's not make changes like this unless we're actually touching this line, we're just making the real history of this line harder to find and making it harder to review for nearly no benefit.

File: Xamarin.PropertyEditing.Mac/Controls/Custom/BaseTextField.cs [hunk 2]
```diff
@@ -0,0 +1,15 @@
+using System;
+using AppKit;
+
+namespace Xamarin.PropertyEditing.Mac
+{
+	public class BasePropertyControlTextField : NSTextField
```
Comment (ermau) [inline Xamarin.PropertyEditing.Mac/Controls/Custom/BaseTextField.cs]: - Does this need to be public?
- Can we try a different name? It's not purely a base if we're using it directly.
- The file name really should match the class name

File: Xamarin.PropertyEditing.Mac/Controls/Custom/UnfocusableTextField.cs [hunk 3]
```diff
@@ -59,14 +59,10 @@ public UnfocusableTextField (CGRect frameRect, string text) : base (frameRect)
...
 		{
-			this.label = new NSTextField {
+			this.label = new BasePropertyControlTextField {
 				AccessibilityElement = false,
 				BackgroundColor = NSColor.Clear,
 				Bordered = false,
-				Cell = {
-					LineBreakMode = NSLineBreakMode.TruncatingTail,
-					UsesSingleLineMode = true,
-				},
```
Comment (ermau) [inline Xamarin.PropertyEditing.Mac/Controls/Custom/UnfocusableTextField.cs]: Shouldn't this have meant the property editor labels would have had a ... and all that? They weren't and still don't, but should.

Comment (CartBlanche) [reply Xamarin.PropertyEditing.Mac/Controls/Custom/UnfocusableTextField.cs]: They do, when clipped "correctly". Look at the `Type` field in the header Panel, when you resize to a narrow width. The `...` is there.

File: Xamarin.PropertyEditing.Mac/Controls/Custom/BaseTextField.cs (hunk 2 above)
Comment (CartBlanche) [reply Xamarin.PropertyEditing.Mac/Controls/Custom/BaseTextField.cs]: Do you have any naming suggestions?
Only other name I can suggest would be one denoting functionality. 
So `EllipsisTextField` ?

File: Xamarin.PropertyEditing.Mac/Controls/Custom/UnfocusableTextField.cs (hunk 3 above)
Comment (ermau) [reply Xamarin.PropertyEditing.Mac/Controls/Custom/UnfocusableTextField.cs]: So we need to fix the clipping separately? 👍

File: Xamarin.PropertyEditing.Mac/Controls/Custom/BaseTextField.cs (hunk 2 above)
Comment (ermau) [reply Xamarin.PropertyEditing.Mac/Controls/Custom/BaseTextField.cs]: Could just be `PropertyTextField`

Comment (CartBlanche) [reply Xamarin.PropertyEditing.Mac/Controls/Custom/BaseTextField.cs]: Sure, why not.

File: Xamarin.PropertyEditing.Mac/Controls/Custom/UnfocusableTextField.cs (hunk 3 above)
Comment (CartBlanche) [reply Xamarin.PropertyEditing.Mac/Controls/Custom/UnfocusableTextField.cs]: Yes. Please see my comment on https://github.com/xamarin/Xamarin.PropertyEditing/pull/529

**Comment by CartBlanche** [comment]
@ermau Changes resolved. New control renamed.

**Comment by CartBlanche** [comment]
@ermau Who would have thought adding constraints would break keyboard navigation. If this is the only thing holding back this PR, I'll revert the change for this control and add an issue specifically to investigate why switching to constraints breaks navigation in this control.

**Comment by ermau** [comment]
Just needs a rebase and good to go.

**Pull request merged**